src/pages/ListPage/CreatePolicyWizard.tsx

```diff
@@ -1,36 +1,56 @@
 import * as React from 'react';
-import { PolicyWizard } from '../../components/Policy/PolicyWizard';
-import { useCreatePolicyMutation, useVerifyPolicyMutation } from '../../services/Api';
+import { PolicyWizard, SavingMode } from '../../components/Policy/PolicyWizard';
+import { useSavePolicyMutation, useVerifyPolicyMutation } from '../../services/Api';
 import * as HttpStatus from 'http-status-codes';
 import { addSuccessNotification } from '../../utils/AlertUtils';
 import { CreatePolicyResponse, VerifyPolicyResponse } from '../../components/Policy/PolicyWizardTypes';
 import { PolicyWithOptionalId } from '../../types/Policy/Policy';
 
-interface CreatePolicyWizardProps {
+type CreatePolicyWizardBase = {
     close: (policyCreated: boolean) => void;
-    isOpen: boolean;
-}
+    initialValue?: PolicyWithOptionalId;
+    savingMode: SavingMode;
+};
+
+type CreatePolicyWizardIsOpen = {
+    isOpen: true;
+} & CreatePolicyWizardBase;
+
+type CreatePolicyWizardIsClose = {
+    isOpen: false;
+} & Partial<CreatePolicyWizardBase>;
+
+type CreatePolicyWizardProps = CreatePolicyWizardIsClose | CreatePolicyWizardIsOpen;
 
 export const CreatePolicyWizard: React.FunctionComponent<CreatePolicyWizardProps> = (props) => {
-    const saveMutation = useCreatePolicyMutation();
+    const saveMutation = useSavePolicyMutation();
     const verifyMutation = useVerifyPolicyMutation();
 
+    if (props.savingMode === SavingMode.UPDATE && props.initialValue?.id === undefined) {
+        throw new Error('Invalid SavingMode UPDATE for initialValue. initialValue must provide a Policy with an id');
+    }
+
     const onSave = (policy: PolicyWithOptionalId): Promise<CreatePolicyResponse> => {
         return saveMutation.mutate(policy).then((res) => {
             switch (res.status) {
                 case HttpStatus.CREATED:
-                    addSuccessNotification('Created', `Policy "${policy.name}" created`);
-                    props.close(true);
+                case HttpStatus.OK:
+                    if (policy.id === undefined) {
+                        addSuccessNotification('Created', `Policy "${policy.name}" created`);
+                    } else {
+                        addSuccessNotification('Saved', `Policy "${policy.name}" has been updated`);
+                    }
+
+                    props.close && props.close(true);
                     return {
                         created: true
                     };
```

Comment:
I think it would be good to put a 'break' statement in here - even if the return prevents the fall-through to make it visually more obvious.